src/Modules/LearningTasks/Tutor.LearningTasks.Core/UseCases/Learning/TaskProgressService.cs

```diff
@@ -55,6 +62,8 @@ public Result<TaskProgressDto> ViewStep(int unitId, int id, int stepId, int lear
 
         var taskProgress = result.Value;
         taskProgress.ViewStep(stepId);
+        _progressRepository.UpdateEvents(taskProgress);
+
         return Update(taskProgress);
```

Comment:
Will this generate two events (one for the `UpdateEvents` and another for `Update` in line 67?

Reply:
Check if this method specifically removes any old events from existence. I do not think this will be the case, but better safe than sorry.

Reply:
It does not save any events without _progressRepository.UpdateEvents(taskProgress);.